core/app/models/workarea/checkout.rb

```diff
@@ -161,8 +161,10 @@ def continue_as(user)
     # @return [self]
     #
     def update(params = {})
-      steps.reduce(true) do |_, step|
-        step.new(self).update(params)
+      return true if params.blank?
+
+      steps.reduce(true) do |result, step|
+        result && step.new(self).update(params)
```

Comment:
This breaks the test I wrote, in that `#update` never returns `true` and won't even get to the additional steps with this code added in. I wanted to showcase it here so we can start discussing a way around the problem. Part of this is that the return values of these methods were never really considered before, so we're shoehorning in this new functionality without really taking a step back and seeing if this is the right approach. I personally have and always will advocate for a separate system for bubbling up and tracking errors in the checkout, such as an ActiveModel::Errors object that just collects errors from all of the models in checkout. If we did things this way, the `#update` method could return back to its original form (by just running an `#each` loop), then we can just check `errors.any?` at the end to determine whether the update succeeded. No errors = update success. Let's stop messing around with this stupid boolean logic and just fix the problem already.

Reply:
Hmm, the struggle is real. Boolean short-circuiting might be an issue as-written where it won't bother to run the second part of the `&&` when the first part is already `false`.

Maybe
```
result = true
steps.each { |s| result &= s.new(self).update(params) }
result
```

`&=` won't short circuit the way `&&` will:
```
irb(main):013:0> x = Proc.new { puts "HI"; true; }
=> #<Proc:0x00007ff59da70518@(irb):13>
irb(main):014:0> y = false
=> false
irb(main):015:0> y && x.call
=> false
irb(main):016:0> y &= x.call
HI
=> false
```

Reply:
that worked!